docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,194 @@
+# Semantic Conventions for Messaging Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [Common attributes](#common-attributes)
+- [Producer metrics](#producer-metrics)
+  * [Metric: `messaging.publish.duration`](#metric-messagingpublishduration)
+  * [Metric: `messaging.publish.messages`](#metric-messagingpublishmessages)
+- [Consumer metrics](#consumer-metrics)
+  * [Metric: `messaging.receive.duration`](#metric-messagingreceiveduration)
+  * [Metric: `messaging.receive.messages`](#metric-messagingreceivemessages)
+  * [Metric: `messaging.deliver.duration`](#metric-messagingdeliverduration)
+  * [Metric: `messaging.deliver.messages`](#metric-messagingdelivermessages)
+
+<!-- tocstop -->
+
+## Common attributes
+
+All messaging metric share the same set of attributes:
+
+<!-- semconv metric.messaging.attributes(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `error.type` | string | Describes a class of error the operation ended with. [1] | `amqp:decode-error`; `KAFKA_STORAGE_ERROR`; `channel-error` | Conditionally Required: [2] |
+| [`messaging.destination.name`](../attributes-registry/messaging.md) | string | The message destination name [3] | `MyQueue`; `MyTopic` | Conditionally Required: [4] |
+| [`messaging.destination.template`](../attributes-registry/messaging.md) | string | Low cardinality representation of the messaging destination name [5] | `/customers/{customerId}` | Conditionally Required: if available. |
+| [`messaging.system`](../attributes-registry/messaging.md) | string | A string identifying the messaging system. | `kafka`; `rabbitmq`; `rocketmq`; `activemq`; `AmazonSQS` | Required |
+| [`network.protocol.name`](../attributes-registry/network.md) | string | [OSI application layer](https://osi-model.com/application-layer/) or non-OSI equivalent. [6] | `amqp`; `mqtt` | Recommended |
+| [`network.protocol.version`](../attributes-registry/network.md) | string | Version of the protocol specified in `network.protocol.name`. [7] | `3.1.1` | Recommended |
+| [`network.transport`](../attributes-registry/network.md) | string | [OSI transport layer](https://osi-model.com/transport-layer/) or [inter-process communication method](https://wikipedia.org/wiki/Inter-process_communication). [8] | `tcp`; `udp` | Recommended |
+| [`network.type`](../attributes-registry/network.md) | string | [OSI network layer](https://osi-model.com/network-layer/) or non-OSI equivalent. [9] | `ipv4`; `ipv6` | Recommended |
```

Comment:
Do we see value in having those dimensions on metrics? We don't have them on HTTP metrics.

Reply:
good point, removed

Reply:
we should get back to it in the scope of https://github.com/open-telemetry/opentelemetry-specification/issues/963 and review anything network-related we put on logical spans/metrics